src/app/core/config-ui/config-field/config-field.component.ts

```diff
@@ -0,0 +1,284 @@
+      innerDataType: [this.entitySchemaFieldWithId.innerDataType],
+
+      defaultValue: [this.entitySchemaFieldWithId.defaultValue],
+      searchable: [this.entitySchemaFieldWithId.searchable],
+      anonymize: [this.entitySchemaFieldWithId.anonymize],
+      //viewComponent: [],
+      //editComponent: [],
+      //showInDetailsView: [],
+      //generateIndex: [],
+      validators: [this.entitySchemaFieldWithId.validators],
+    });
+
+    this.updateDataTypeAdditional(this.form.get("dataType").value);
+    this.form
+      .get("dataType")
+      .valueChanges.subscribe((v) => this.updateDataTypeAdditional(v));
+    this.updateForNewOrExistingField();
+  }
+
+  private updateForNewOrExistingField() {
+    if (!!this.entitySchemaFieldWithId.id) {
+      // existing fields' id is readonly
+      this.formId.disable();
+    } else {
+      const autoGenerateSubscr = this.form
+        .get("label")
+        .valueChanges.subscribe((v) => this.autoGenerateId(v));
+      // stop updating id when user manually edits
+      this.formId.valueChanges.subscribe(() =>
+        autoGenerateSubscr.unsubscribe(),
+      );
+    }
+  }
+  private autoGenerateId(updatedLabel: string) {
+    const generatedId = generateSimplifiedId(updatedLabel);
+    this.formId.setValue(generatedId, { emitEvent: false });
+  }
+
+  private initAvailableDatatypes(dataTypes: DefaultDatatype[]) {
+    this.dataTypes = dataTypes
+      .filter((d) => d.label !== DefaultDatatype.label) // hide "internal" technical dataTypes that did not define a human-readable label
+      .map((d) => ({
+        label: d.label,
+        value: d.dataType,
+      }));
+  }
+  objectToLabel = (v: SimpleDropdownValue) => v?.label;
+  objectToValue = (v: SimpleDropdownValue) => v?.value;
+  createNewAdditionalOption: (input: string) => SimpleDropdownValue;
+
+  private updateDataTypeAdditional(dataType: string) {
+    this.resetAdditional();
+
+    if (dataType === ConfigurableEnumDatatype.dataType) {
+      this.initAdditionalForEnum();
+    } else if (
+      dataType === EntityDatatype.dataType ||
+      dataType === EntityArrayDatatype.dataType
+    ) {
+      this.initAdditionalForEntityRef();
+    }
+
+    // hasInnerType: [ArrayDatatype.dataType].includes(d.dataType),
+
+    // TODO: this mapping of having an "additional" schema should probably become part of Datatype classes
+  }
+
+  private initAdditionalForEnum() {
+    this.typeAdditionalOptions = this.configurableEnumService
+      .listEnums()
+      .map((x) => ({
+        label: Entity.extractEntityIdFromId(x),
+        value: Entity.extractEntityIdFromId(x),
+      }));
+    this.formAdditional.addValidators(Validators.required);
+
+    this.createNewAdditionalOption = (text) => {
+      const newOption = {
+        value: text,
+        label: $localize`[new options set]: "${text}"`,
+        isNew: true,
+      };
+
+      setTimeout(() => {
+        // only offer one newly created configurable-enum id
```

Comment:
```suggestion
        // when the user creates a new configurable-enum id (instead of selecting a pre-existing enum), remove previously generated new configurable-enum id (because at most one newly created enum can be used)
```